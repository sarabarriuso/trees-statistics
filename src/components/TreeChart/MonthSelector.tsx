import Slider from '@mui/material/Slider';

interface IMonthSelectorProps {
  description: string;
  defaultMonth: number;
  monthSelectedCallback: (value: number) => void;
}
const MonthSelector: React.FC<IMonthSelectorProps> = ({
  description,
  defaultMonth,
  monthSelectedCallback,
}) => {
  const DEFAULT_CLASSNAME = 'month-selector';

  const marks = [
    {
      value: 1,
      label: 'Jan',
    },
    {
      value: 2,
      label: 'Feb',
    },
    {
      value: 3,
      label: 'Mar',
    },
    {
      value: 4,
      label: 'Apr',
    },
    {
      value: 5,
      label: 'May',
    },
    {
      value: 6,
      label: 'Jun',
    },
    {
      value: 7,
      label: 'Jul',
    },
    {
      value: 8,
      label: 'Aug',
    },
    {
      value: 9,
      label: 'Sep',
    },
    {
      value: 10,
      label: 'Oct',
    },
    {
      value: 11,
      label: 'Nov',
    },
    {
      value: 12,
      label: 'Dec',
    },
  ];

  const handleChange = (
    _event: Event | React.SyntheticEvent<Element, Event>,
    value: number | number[],
  ) => {
    monthSelectedCallback(value as number);
  };

  return (
    <div className={DEFAULT_CLASSNAME}>
      <p>{description}</p>
      <Slider
        aria-label='Day'
        defaultValue={defaultMonth}
        track={false}
        valueLabelDisplay='auto'
        step={1}
        marks={marks}
        min={1}
        max={12}
        onChangeCommitted={handleChange}
      />
    </div>
  );
};

export default MonthSelector;
