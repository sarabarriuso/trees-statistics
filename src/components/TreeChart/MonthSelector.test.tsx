import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MonthSelector from './MonthSelector';
import { Slider } from '@mui/material';

export default describe('Month Selector Index Tests', () => {
  it('Renders successfully ', () => {
    // Arrange
    // Act
    const tree = shallow(
      <MonthSelector
        description='Hi I am a description'
        defaultMonth={1}
        monthSelectedCallback={() => {
          return;
        }}
      />,
    );

    //Assert
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('Changing the month will trigger the monthSelectedCallback', () => {
    // Arrange
    let callbackCalled = false;

    // Act
    const tree = shallow(
      <MonthSelector
        description='Hi I am a description'
        defaultMonth={1}
        monthSelectedCallback={() => {
          callbackCalled = true;
        }}
      />,
    );

    const slider = tree.find(Slider);
    const sliderCallback = slider.props().onChangeCommitted;
    if (sliderCallback !== undefined) {
      sliderCallback(new Event('click', undefined), 5);

      //Assert
      expect(callbackCalled).toBeTruthy();
    }
  });
});
