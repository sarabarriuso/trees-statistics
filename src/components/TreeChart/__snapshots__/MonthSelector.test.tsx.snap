// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month Selector Index Tests Renders successfully  1`] = `
<div
  className="month-selector"
>
  <p>
    Hi I am a description
  </p>
  <ForwardRef(Slider)
    aria-label="Day"
    defaultValue={1}
    marks={
      Array [
        Object {
          "label": "Jan",
          "value": 1,
        },
        Object {
          "label": "Feb",
          "value": 2,
        },
        Object {
          "label": "Mar",
          "value": 3,
        },
        Object {
          "label": "Apr",
          "value": 4,
        },
        Object {
          "label": "May",
          "value": 5,
        },
        Object {
          "label": "Jun",
          "value": 6,
        },
        Object {
          "label": "Jul",
          "value": 7,
        },
        Object {
          "label": "Aug",
          "value": 8,
        },
        Object {
          "label": "Sep",
          "value": 9,
        },
        Object {
          "label": "Oct",
          "value": 10,
        },
        Object {
          "label": "Nov",
          "value": 11,
        },
        Object {
          "label": "Dec",
          "value": 12,
        },
      ]
    }
    max={12}
    min={1}
    onChangeCommitted={[Function]}
    step={1}
    track={false}
    valueLabelDisplay="auto"
  />
</div>
`;
